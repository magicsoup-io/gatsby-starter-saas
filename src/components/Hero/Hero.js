import React from 'react'
import { StaticQuery, graphql } from 'gatsby'
import { Flex, Box, Card, Heading } from '@magicsoup.io/stock'
import { P } from '../../styled'

const Hero = () => (
  <StaticQuery
    query={graphql`
      query HeroQuery {
        heroJson {
          displayTitle
          subTitle
          github
        }
      }
    `}
    render={data => (
      <Card variant='grey'>
        <Flex
          p={4}
          pt={[6,7]}
          alignItems='center'
        >
          <Box>
            <Heading as='h1' variant='h1' fontSize={[6,7,8]} m={0}>
              {data.heroJson.displayTitle}
            </Heading>
            <P fontSize={3} mb={3}>
              {data.heroJson.subTitle}
            </P>
            <a 
              className='github-button' 
              data-icon='octicon-star'
              href={data.heroJson.github} 
              data-size='large' 
              data-show-count='true' 
              aria-label={`Star this repo on GitHub`}>
              Star
            </a>
          </Box>
        </Flex>
      </Card>
    )}
  />
)

export default Hero
