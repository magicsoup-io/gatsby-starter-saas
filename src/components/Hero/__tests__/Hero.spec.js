import React from "react"
import renderer from "react-test-renderer"
import { StaticQuery } from 'gatsby'
import Hero from "../Hero"

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      heroJson: {
        displayTitle: 'Display title',
        subTitle: 'Subtitle',
      },
    })
  )
})

describe("Hero", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Hero />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})