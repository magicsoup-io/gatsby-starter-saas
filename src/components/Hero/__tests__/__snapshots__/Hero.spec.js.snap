// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<div
  className="sc-bdVaJa sc-htpNat jocheC"
>
  <div
    className="sc-bdVaJa sc-ifAKCX jYmEKF"
  >
    <div
      className="sc-bdVaJa eAkHCv"
    >
      <h1
        className="sc-bdVaJa sc-EHOje sc-bZQynM kMuPVV"
        fontSize={
          Array [
            6,
            7,
            8,
          ]
        }
      >
        Display title
      </h1>
      <p
        className="sc-bdVaJa sc-EHOje sc-gqjmRU cwURln"
        fontSize={3}
      >
        Subtitle
      </p>
      <a
        aria-label="Star this repo on GitHub"
        className="github-button"
        data-icon="octicon-star"
        data-show-count="true"
        data-size="large"
      >
        Star
      </a>
    </div>
  </div>
</div>
`;
